### PR TITLE
 #159569195 Users can change default center

### PR DESCRIPTION
**Pull request by PaulKariukiRimiru**
**What does this PR do?**

This PR adds functionality to allow a user to change their default center from the profile dropdown

**Description of Task to be completed?**

Given that I am a user logged into Activo, I should be able to navigate to my settings page and change my default center

**How should this be manually tested?**

After cloning and running the branch click on the drop-down icon on the far right of the navigation bar, a dropdown menu will appear. Click the `change` button and a modal with a list of centers will appear. Select one center and click save changes.

**Any background context you want to provide?**

- The endpoint to persist this feature in the backend is not ready, therefore I had to design a module user that will contain the current logged in user details. I would like to ask for feedback on this module in order to improve it.

**What are the relevant pivotal tracker stories?**

[#159569195 As a user, I should be able to set my default center](https://www.pivotaltracker.com/story/show/159569195)

**Screenshots (if appropriate)**

screen shot 2018-08-13 at 10 42 05

screen shot 2018-08-13 at 10 44 06

**Questions:**

Suggestion on how the user information that is not provided in the token be stored in the state?

### CODE REVIEW

File: src/components/NavBar/index.tsx [hunk 1]
```diff
@@ -91,15 +118,44 @@ export class NavBar extends React.Component<NavBarProps, {}> {
...
 
+  /**
+   * This method is called when a user changes their default center
+   * 
+   * @memberof NavBar
+   * @param {string} selectedCenter
+   * @returns {void}
+   */
+  onSaveSelectedCenter = (selectedCenter) => {
+    this.props.editUserCenter(selectedCenter);
```
Comment (andela-kakpobome) [inline src/components/NavBar/index.tsx]: This is supposed to take two arguments, but you only pass one, hence the dispatched action won't be able to pass the proper userId to save the center for.

File: src/components/NavBar/index.tsx [hunk 2]
```diff
@@ -154,7 +214,15 @@ export class NavBar extends React.Component<NavBarProps, {}> {
...
-  userDetails: state.auth.userDetail,
+  availableCenters: state.centers.data,
+  user: { ...state.user.userDetails.UserInfo, center: state.user.userDetails.center },
+});
+
+export const mapDispatchToProps = dispatch => ({
+  logoutUser: () => dispatch(logoutUser()),
+  getCenters: () => dispatch(getCenters()),
+  getUserDetails: userId => dispatch(getUserDetails(userId)),
+  editUserCenter: (userCenter, userId) => dispatch(editUserCenter(userId, userCenter)),
```
Comment (andela-kakpobome) [inline src/components/NavBar/index.tsx]: Let the arguments be aligned, if `userId` is the first argument, let it be the first for both the method on props and the thunk, switching them as you've done here is somewhat confusing to read.

File: src/components/NavBar/NavBar.test.tsx [hunk 3]
```diff
@@ -2,31 +2,72 @@
...
+        },
+      ],
+    });
+    const instance = wrapper.instance() as NavBar;
+    instance.onSaveSelectedCenter({
+      name: 'test',
+      id: 1,
+    });
+
+    expect(wrapper.instance().props.editUserCenter).toBeCalledWith({
```
Comment (andela-kakpobome) [inline src/components/NavBar/NavBar.test.tsx]: Shouldn't you also test that it get's called with the userId, not just the center? The thunk that this prop is mapped to expects those two arguments.

**Comment by andela-kakpobome** [comment]
For some reason, you did not include the changes to address the feedback about the dropdown having discrepancies with the design. I've had to implement that myself. Do well to go through the changes I made and take note as feedback for subsequent PRs. Some key things to note:

- There's almost no scenario where you should be giving a `p` element a width and height(even though the values exist in the design spec), font-size would suffice.

- Instead of giving several elements within a container, separate margin top values, you should instead, give the container the margin top

- With margin/padding, work your way from the outside in, instead of giving each element a margin and padding, trying to get them to fit, give the container a padding to push it's content inward, in an equidistant fashion.

**Comment by PaulKariukiRimiru** [comment]
Thanks @andela-kakpobome , I have looked at the changes you added and understood what you meant on the discrepancies. Ill take note of them in future tasks.

File: src/components/CentersModal/CentersModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        {
+          id: '1',
+          name: 'test',
+          isSelected: false,
+        },
+      ],
+    });
+    wrapper.find(Button).first().simulate('click');
+    wrapper.find('.center-modal-footer__save-changes').simulate('click');
+    expect(wrapper.props().onSaveSelectedCenter).toBeCalledWith('1', '1');
```
Comment (andela-kakpobome) [inline src/components/CentersModal/CentersModal.test.tsx]: Use the variable directly instead i.e props.userDetails.id etc

File: src/components/NavBar/NavBar.test.tsx [hunk 5]
```diff
@@ -2,31 +2,66 @@
...
+        {
+          name: 'test',
+          id: '1',
+        },
+      ],
+    });
+    const instance = wrapper.instance() as NavBar;
+    instance.onSaveSelectedCenter('1');
+
+    expect(wrapper.instance().props.editUserCenter).toBeCalledWith('1', '1');
```
Comment (andela-kakpobome) [inline src/components/NavBar/NavBar.test.tsx]: Use variables instead, it's more readable.

**Pull request merged**